llama.cpp

```diff
@@ -1828,19 +1826,149 @@ struct llama_model_loader {
+
+    template<typename TI, typename TO>
+    void gk_set_lit(TI i, TO o) {
+        (void)i; (void)o;
+        throw std::runtime_error(format("gk_set_lit can't handle types: in=%s, out=%s",
+            typeid(TI).name(), typeid(TO).name()));
+    }
+
+    template<typename T>
+    typename std::enable_if<std::is_integral<T>::value, void>::type
+    gk_set_lit(const int64_t & i, T & o) { o = T(i); }
+
+    template<typename T>
+    typename std::enable_if<std::is_floating_point<T>::value, void>::type
+    gk_set_lit(const double & i, T & o) { o = T(i); }
+
+    template<typename T>
+    void gk_set_lit(const T & i, T & o) { o = i; }
+
+    public:
+    template<typename T>
+    bool get_key(const std::string & key, T & result, const bool required = false) {
+        const auto & tt = typeid(T);
+        enum gguf_type gt = GGUF_TYPE_COUNT;
+        enum llama_model_kv_override_type ot = LLAMA_KV_OVERRIDE_INT;
+        bool is_signed = false, can_override = true;
+        if (tt == typeid(uint8_t)) {
+            gt = GGUF_TYPE_UINT8;
+        } else if (tt == typeid(uint16_t)) {
+            gt = GGUF_TYPE_UINT16;
+        } else if (tt == typeid(uint32_t)) {
+            gt = GGUF_TYPE_UINT32;
+        } else if (tt == typeid(uint64_t)) {
+            gt = GGUF_TYPE_UINT64;
+        } else if (tt == typeid(int8_t)) {
+            is_signed = true;
+            gt = GGUF_TYPE_INT8;
+        } else if (tt == typeid(int16_t)) {
+            is_signed = true;
+            gt = GGUF_TYPE_INT16;
+        } else if (tt == typeid(int32_t)) {
+            is_signed = true;
+            gt = GGUF_TYPE_INT32;
+        } else if (tt == typeid(int64_t)) {
+            is_signed = true;
+            gt = GGUF_TYPE_INT64;
+        } else if (tt == typeid(float)) {
+            is_signed = true;
+            gt = GGUF_TYPE_FLOAT32;
+            ot = LLAMA_KV_OVERRIDE_FLOAT;
+        } else if (tt == typeid(double)) {
+            is_signed = true;
+            gt = GGUF_TYPE_FLOAT64;
+            ot = LLAMA_KV_OVERRIDE_FLOAT;
+        } else if (tt == typeid(bool)) {
+            gt = GGUF_TYPE_BOOL;
+            ot = LLAMA_KV_OVERRIDE_BOOL;
+        } else if (tt == typeid(std::string)) {
+            can_override = false;
+            gt = GGUF_TYPE_STRING;
+        } else {
+            throw std::runtime_error(format("request for key '%s' with unknown result type: %s", key.c_str(), tt.name()));
+        }
+
+        if (can_override) {
+            auto it = kv_overrides.find(key);
+            if (it != kv_overrides.end()) {
+                struct llama_model_kv_override & po = it->second;
+                if (po.tag != ot) {
+                    // Bad type
+                } else if (ot == LLAMA_KV_OVERRIDE_INT && po.int_value < 0 && !is_signed) {
+                    // Out of range
+                } else {
+                    switch (po.tag) {
+                        case LLAMA_KV_OVERRIDE_INT: gk_set_lit(po.int_value, result); break;
```

Comment:
These call the deleted version because result is the wrong type for two of them. Since you know the type of T, you should dispatch at compile time to the correct function (e.g. by accessing the union member within a template function), not using a switch at runtime.